scripts/irods/test/test_dynamic_peps.py

```diff
@@ -492,3 +492,67 @@ def test_if_Genquery2Input_is_exposed__issue_7570(self):
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_1', self.user.zone_name])
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_2', '0'])
                 self.user.run_icommand(['imeta', 'rm', '-C', self.user.session_collection, f'{avu_prefix}_3', '0'])
+
+    @unittest.skipIf(plugin_name != 'irods_rule_engine_plugin-irods_rule_language' or test.settings.RUN_IN_TOPOLOGY, "Requires NREP and single node zone.")
+    def test_if_StructFileExtAndRegInp_is_exposed__issue_7413(self):
+        with temporary_core_file() as core:
+            avu_prefix = 'test_if_StructFileExtAndRegInp_is_exposed__issue_7413'
+
+            # Add a rule to core.re which when triggered will add AVUs to the user's
+            # session collection.
+            core.add_rule(dedent('''
+            pep_api_struct_file_ext_and_reg_post(*INSTANCE, *COMM, *STRUCTINP) {{
+                *v = *STRUCTINP.objPath;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_0', *v, '');
+
+                *v = *STRUCTINP.collection_path;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_1', *v, '');
+
+                *v = *STRUCTINP.oprType;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_2', *v, '');
+
+                *v = *STRUCTINP.flags;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_3', *v, '');
+
+                *v = *STRUCTINP.defRescName;
+                msiModAVUMetadata('-C', '{self.user.session_collection}', 'add', '{avu_prefix}_4', *v, '');
+            }}
+            '''.format(**locals())))
+
+            TEST_COLLECTION_NAME = "%s/%s" % (self.user.session_collection, 'testcoll')
+            TEST_TARFILE_NAME = "%s/%s" % (self.user.session_collection, 'test.tar')
+
+            try:
+                self.user.assert_icommand(['imkdir', TEST_COLLECTION_NAME], 'STDOUT')
+                # Make some test files...
+                for x in range(0, 5):
+                    self.user.assert_icommand(['itouch', "%s/file%s" % (TEST_COLLECTION_NAME, x)])
+
+                self.user.assert_icommand(['ibun', '-cDtar', "%s/%s" % (self.user.session_collection, 'test.tar'), TEST_COLLECTION_NAME])
+
+                self.user.assert_icommand(['imkdir', TEST_COLLECTION_NAME + "_backup"], 'STDOUT')
+
+                # Should trigger PEP
+                self.user.run_icommand(['ibun', '-x', TEST_TARFILE_NAME, TEST_COLLECTION_NAME + "_backup"])
```

Comment:
`run_icommand` -> `assert_icommand`

Reply:
I'm not sure why, but this command returns nonzero. When we were testing this in discush last week, we saw a nonzero return despite it still working as expected.
When this line is as-is, it passes. Otherwise, I think it errors.

Reply:
Hmm. We definitely have tests around `ibun -x`.

Have you tried running `ibun -x` manually and seeing what is returned by `$?`?

Reply:
Right, we should confirm that `ibun -x` behaves the same way before and after this change. I don't remember seeing that in our discussion (sorry) but that doesn't seem like what we want. If that is how `ibun -x` behaves before these changes, then we should make a separate issue to look into that later.

Reply:
Huh, false alarm. Just re-ran it with assert and it seems ok now?

Reply:
Well, that's good. It shall come out in testing in the run up to release if not. Hooray